com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/FileService.java

```diff
@@ -1,269 +1,309 @@
+     * @param is input stream to read the content from
+     * @return an object representing the file upload
+     * @throws FileStorageException
+     */
+    public FileEntry addFile(String space, String namespace, String name,
+        FileUploadProcessor<? extends OutputStream, ? extends OutputStream> fileInfoProcessor, InputStream is) throws FileStorageException {
+        // Stream the file to a temp location and get the size and MD5 digest
+        // as an alternative we can pass the original stream to the database,
+        // and decorate the blob stream to calculate digest and size, but this will still require
+        // two roundtrips to the database (insert of the content and then update with the digest and
+        // size), which is probably inefficient
+        FileInfo fileInfo = null;
+        FileEntry fileEntry = null;
+        try (InputStream inputStream = is) {
+            fileInfo = FileUploader.uploadFile(inputStream, fileInfoProcessor);
+            fileEntry = addFile(space, namespace, name, fileInfoProcessor, fileInfo);
+        } catch (IOException e) {
+            logger.debug(e.getMessage(), e);
+        } finally {
+            if (fileInfo != null) {
+                FileUploader.removeFile(fileInfo);
+            }
+        }
+        return fileEntry;
+    }
+
+    public FileEntry addFile(String space, String namespace, String name,
+        FileUploadProcessor<? extends OutputStream, ? extends OutputStream> fileInfoProcessor, File existingFile)
+        throws FileStorageException {
+        try {
+            FileInfo fileInfo = createFileInfo(existingFile);
+
+            return addFile(space, namespace, name, fileInfoProcessor, fileInfo);
+        } catch (NoSuchAlgorithmException e) {
+            throw new SLException(Messages.ERROR_CALCULATING_FILE_DIGEST, existingFile.getName(), e);
+        } catch (FileNotFoundException e) {
+            throw new FileStorageException(MessageFormat.format(Messages.ERROR_FINDING_FILE_TO_UPLOAD, existingFile.getName()), e);
+        } catch (IOException e) {
+            throw new FileStorageException(MessageFormat.format(Messages.ERROR_READING_FILE_CONTENT, existingFile.getName()), e);
+        }
+    }
+
+    public List<FileEntry> listFiles(final String space, final String namespace) throws FileStorageException {
+        try {
+            return getSqlQueryExecutor().execute(getSqlFileQueryProvider().getListFilesQuery(space, namespace));
+        } catch (SQLException e) {
+            throw new FileStorageException(MessageFormat.format(Messages.ERROR_GETTING_FILES_WITH_SPACE_AND_NAMESPACE, space, namespace),
+                e);
+        }
+    }
+
+    public List<FileEntry> listFiles(final String namespace) throws FileStorageException {
+        return listFiles(null, namespace);
+    }
+
+    public FileEntry getFile(final String space, final String id) throws FileStorageException {
+        FileEntry fileEntry = getFileEntry(space, id);

```

Comment:
Return directly.

Reply:
Done. :)

Reply:
Fixed. Also removed the protected method since getFile works the same way in DatabaseFileService and FileService.